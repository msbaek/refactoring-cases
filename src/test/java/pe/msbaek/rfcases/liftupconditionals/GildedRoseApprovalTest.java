package pe.msbaek.rfcases.liftupconditionals;

import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

@UseReporter(DiffReporter.class)
public class GildedRoseApprovalTest {
    @Test
    public void foo() {
        String [] name = {"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"};
        Integer [] sellIn = {-1, 0, 11};
        Integer [] quality = {0, 1, 49, 50};

        CombinationApprovals.verifyAllCombinations(this::getItems, name, sellIn, quality);
    }

    private String getItems(String name, int sellIn, int quality) {
        GildedRose app = new GildedRose(new Item[]{
                new Item(name, sellIn, quality)
        });
        app.updateQuality();
        return Arrays.asList(app.items).toString();
    }
}