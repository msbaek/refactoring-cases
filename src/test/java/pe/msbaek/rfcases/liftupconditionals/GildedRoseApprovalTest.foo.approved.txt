[foo, -1, 0] => [foo, -2, 0] 
[foo, -1, 1] => [foo, -2, 0] 
[foo, -1, 49] => [foo, -2, 47] 
[foo, -1, 50] => [foo, -2, 48] 
[foo, 0, 0] => [foo, -1, 0] 
[foo, 0, 1] => [foo, -1, 0] 
[foo, 0, 49] => [foo, -1, 47] 
[foo, 0, 50] => [foo, -1, 48] 
[foo, 11, 0] => [foo, 10, 0] 
[foo, 11, 1] => [foo, 10, 0] 
[foo, 11, 49] => [foo, 10, 48] 
[foo, 11, 50] => [foo, 10, 49] 
[Aged Brie, -1, 0] => [Aged Brie, -2, 2] 
[Aged Brie, -1, 1] => [Aged Brie, -2, 3] 
[Aged Brie, -1, 49] => [Aged Brie, -2, 50] 
[Aged Brie, -1, 50] => [Aged Brie, -2, 50] 
[Aged Brie, 0, 0] => [Aged Brie, -1, 2] 
[Aged Brie, 0, 1] => [Aged Brie, -1, 3] 
[Aged Brie, 0, 49] => [Aged Brie, -1, 50] 
[Aged Brie, 0, 50] => [Aged Brie, -1, 50] 
[Aged Brie, 11, 0] => [Aged Brie, 10, 1] 
[Aged Brie, 11, 1] => [Aged Brie, 10, 2] 
[Aged Brie, 11, 49] => [Aged Brie, 10, 50] 
[Aged Brie, 11, 50] => [Aged Brie, 10, 50] 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => [Backstage passes to a TAFKAL80ETC concert, -2, 0] 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => [Backstage passes to a TAFKAL80ETC concert, -2, 0] 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => [Backstage passes to a TAFKAL80ETC concert, -2, 0] 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => [Backstage passes to a TAFKAL80ETC concert, -2, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => [Backstage passes to a TAFKAL80ETC concert, -1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => [Backstage passes to a TAFKAL80ETC concert, -1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => [Backstage passes to a TAFKAL80ETC concert, -1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => [Backstage passes to a TAFKAL80ETC concert, -1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => [Backstage passes to a TAFKAL80ETC concert, 10, 1] 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => [Backstage passes to a TAFKAL80ETC concert, 10, 2] 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => [Backstage passes to a TAFKAL80ETC concert, 10, 50] 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => [Backstage passes to a TAFKAL80ETC concert, 10, 50] 
[Sulfuras, Hand of Ragnaros, -1, 0] => [Sulfuras, Hand of Ragnaros, -1, 0] 
[Sulfuras, Hand of Ragnaros, -1, 1] => [Sulfuras, Hand of Ragnaros, -1, 1] 
[Sulfuras, Hand of Ragnaros, -1, 49] => [Sulfuras, Hand of Ragnaros, -1, 49] 
[Sulfuras, Hand of Ragnaros, -1, 50] => [Sulfuras, Hand of Ragnaros, -1, 50] 
[Sulfuras, Hand of Ragnaros, 0, 0] => [Sulfuras, Hand of Ragnaros, 0, 0] 
[Sulfuras, Hand of Ragnaros, 0, 1] => [Sulfuras, Hand of Ragnaros, 0, 1] 
[Sulfuras, Hand of Ragnaros, 0, 49] => [Sulfuras, Hand of Ragnaros, 0, 49] 
[Sulfuras, Hand of Ragnaros, 0, 50] => [Sulfuras, Hand of Ragnaros, 0, 50] 
[Sulfuras, Hand of Ragnaros, 11, 0] => [Sulfuras, Hand of Ragnaros, 11, 0] 
[Sulfuras, Hand of Ragnaros, 11, 1] => [Sulfuras, Hand of Ragnaros, 11, 1] 
[Sulfuras, Hand of Ragnaros, 11, 49] => [Sulfuras, Hand of Ragnaros, 11, 49] 
[Sulfuras, Hand of Ragnaros, 11, 50] => [Sulfuras, Hand of Ragnaros, 11, 50] 
